report.php

```diff
@@ -0,0 +1,50 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Uses the report builder instance imported in classes/reengagement_report.php
+ *
+ * @package    mod_reengagement
+ * @copyright  2021 Catalyst IT
+ * @author     Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+declare(strict_types=1);
+
+use core_reportbuilder\system_report_factory;
+use mod_reengagement\reengagement_system_report;
+
+
+require_once('../../config.php');
+
+require_login();
+
```

Comment:
missing a require_capability call here - which means that anyone logged in can view this report (security issue) - ideally we should create a new capability - maybe call it "mod/reengagement:viewsitereport" and then check that capability right under the require_login() call.

Reply:
Resolved though I had used a slightly different strategy. Please refer to the new PR for it.